src/test/mgr/test_progress.py

```diff
@@ -0,0 +1,212 @@
+#python unit test
+import unittest
+import os
+import sys
+from mock import Mock
+import pytest
+import json
+os.environ['UNITTEST'] = "1"
+sys.path.insert(0, "../../pybind/mgr")
+from progress import module
+
+class TestPgRecoveryEvent(object):
+    # Testing PgRecoveryEvent class
+
+    def setup(self):
+        # Creating the class and Mocking 
+        # a bunch of attributes for testing
+        self.test_class = module.PgRecoveryEvent
+        self._pgs = [module.PgId(1,i) for i in range(3)]
+        self._evacuate_osds = [0]
+        self._original_pg_count = len(self._pgs)
+        self._original_bytes_recovered = None
+        self.module_class = Mock()
+        self.module_class.info = Mock()
+        self._refresh = Mock()
+        self._progress = 0.0
+        self._message = None
+        self._start_epoch = 30
+
+    def test_pg_update(self):
+        # Test for a completed event when the pg states show active+clear
+        pg_dump = {
+                "pg_stats":[
+        {
+          "state": "active+clean",
+          "stat_sum": {
+            "num_bytes": 10,
+            "num_bytes_recovered": 10
+          },
+          "up": [
+            3,
+            1
+          ],
+          "acting": [
+            3,
+            1
+          ],
+          "pgid": "1.0",
+          "reported_epoch": "30"
+        },
+       {
+          "state": "active+clean",
+          "stat_sum": {
+            "num_bytes": 10,
+            "num_bytes_recovered": 10
+          },
+          "up": [
+            3,
+            1
+          ],
+          "acting": [
+            3,
+            1
+          ],
+          "pgid": "1.1",
+          "reported_epoch": "30"
+        },
+       {
+          "state": "active+clean",
+          "stat_sum": {
+            "num_bytes": 10,
+            "num_bytes_recovered": 10
+          },
+          "up": [
+            3,
+            1
+          ],
+          "acting": [
+            3,
+            1
+          ],
+          "pgid": "1.2",
+          "reported_epoch": "30"
+        }
+        ]
+        }
+
+        self.test_class.pg_update(self,pg_dump,self.module_class)
```

Comment:
test_class is literally a class object here, not an instance of the class, so this will fail since pg_update() requires an instance (since it has self as the first parameter)